### PR TITLE
[Github] Add steps to build clang docs to CI

### PR DESCRIPTION
**Pull request by boomanaiden154**
Recently, support for building the LLVM documentation within Github actions landed, allowing for easy testing of the docs both pre and post landing. This patch extends that functionality to clang and adds in additional support to the docs Github workflow to only build the docs for the subproject whose documentation has been touched.

### CODE REVIEW

**Comment by boomanaiden154** [comment]
I've set everything up in the same job for now mostly because it was simple. We can split it into multiple jobs if necessary, but I don't think many patches touch the docs of multiple subprojects, and those that do still won't run into any problems with the job taking too long. From my testing on my fork, building both the LLVM and clang docs takes under 10 minutes, even with clang having to build `clang-tblgen`.

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -39,10 +41,27 @@ jobs:
...
+            clang:
+              - 'clang/docs/**'
+      - name: Build LLVM docs
+        if: steps.docs-changed-subprojects.outputs.llvm_any_changed == 'true'
         run: |
-          mkdir build
-          cd build
+          mkdir llvm-build
+          cd llvm-build
           cmake -GNinja -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_SPHINX=ON -DSPHINX_OUTPUT_HTML=ON -DSPHINX_OUTPUT_MAN=ON ../llvm
```
Comment (tstellar) [inline .github/workflows/docs.yml:58]: If you pass `-B llvm-build` to cmake, you can skip the `mkdir and cd` steps.

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -39,10 +41,27 @@ jobs:
...
           TZ=UTC ninja docs-llvm-html docs-llvm-man
```
Comment (tstellar) [inline .github/workflows/docs.yml:59]: And then pass `-C llvm-build ` here.

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -39,10 +41,27 @@ jobs:
...
+      - name: Build Clang docs
+        if: steps.docs-changed-subprojects.outputs.clang_any_changed == 'true'
+        run: |
+          mkdir clang-build
+          cd clang-build
+          cmake -GNinja -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS="clang" -DLLVM_ENABLE_SPHINX=ON -DSPHINX_OUTPUT_HTML=ON -DSPHINX_OUTPUT_MAN=ON ../llvm
+          TZ=UTC ninja docs-clang-html docs-clang-man
```
Comment (tstellar) [inline .github/workflows/docs.yml:66]: Same comment here as with llvm pass `-B clang-build` to cmake and `-C clang-build` to ninja.

**Comment by boomanaiden154** [comment]
This will fix https://github.com/llvm/llvm-project/issues/65725.

**Review by tstellar — APPROVED** [review]
Thank you this looks great.  LGTM.

**Comment by boomanaiden154** [comment]
Thank you for the review! I'll probably post something on discourse soon too to get attention for other subprojects that might want this.

**Pull request merged**